### PR TITLE
[DOCS] Added USB DFU and CDC tools guide

### PR DESCRIPTION
**Pull request by pedrominatel**
## Description of Change

This PR includes the guide for the USB peripheral for flashing and monitoring the supported devices in DFU and CDC modes.

## Tests scenarios

N/A

## Related links

N/A

### CODE REVIEW

File: docs/source/tutorials/cdc_dfu_flash.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+========= =======================
+SoC       USB Peripheral Support
+========= =======================
+ESP32-S2  CDC and DFU
+ESP32-C3  CDC only
+ESP32-S3  CDC and DFU
+========= =======================
+
+It's important that your board includes the USB connector attached to the embedded USB from the SoC. If your board doesnt have the USB connector, you can attach an external one to the USB pins.
```
Comment (VojtechBartoska) [inline docs/source/tutorials/cdc_dfu_flash.rst:24]: doesn't

File: docs/source/tutorials/cdc_dfu_flash.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+This instructions will only work on the supported devices with the embedded USB peripheral. The external USB-to-Serial doesn't works on this tutorial.
```
Comment (VojtechBartoska) [inline docs/source/tutorials/cdc_dfu_flash.rst:26]: only work as there is doesn't

File: docs/source/tutorials/cdc_dfu_flash.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+1. Enter into Download Mode manually
+
+This step is done only for the first time you flash the firmware in this mode. To enter into the download mode, you need to press and hold BOOT button and press and release the RESET button.
+
+To check if this procedure was done correctly, now you will see the new USB device listed in the available ports. Select this new device in the **Port** option.
+
+2. Configure the USB DFU
+
+Now you need to set the USB DFU as default on BOOT and for flashing.
```
Comment (VojtechBartoska) [inline docs/source/tutorials/cdc_dfu_flash.rst:51]: In the next step you can set the USB DFU..

(Sentence "now you need" was used before few times so this change can make the guide more readeble)

**Review by VojtechBartoska — APPROVED** [review]
LGTM, just few typos.

I was following the guide and works for me pretty well.

**Comment by me-no-dev** [comment]
good to go?

**Comment by SamchanGuo** [comment]
是先写的英文文档，然后再翻译成中文，这个流程吗？
Was the English document written first, and then translated into Chinese?

**Pull request merged**